### PR TITLE
Code quality concerns

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
/cc @GrabYourPitchforks

### CODE REVIEW

File: src/coreclr/vm/methodtablebuilder.cpp [hunk 1]
```diff
@@ -4178,18 +4178,18 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
 
         if (!fIsStatic)
         {
-            pFD = &pFieldDescList[dwCurrentDeclaredField];
+            pFD = &pFieldDescList[dwCurrentDeclaredField]; // lgtm [cpp/upcast-array-pointer-arithmetic]
```
Comment (GrabYourPitchforks) [inline src/coreclr/vm/methodtablebuilder.cpp:4181]: All suppressions require a justification of 25+ characters.

Example:
```cpp
pFD = &pFieldDescList[dwCurrentDeclaredField]; // lgtm [cpp/upcast-array-pointer-arithmetic] - Some text here that's 25 characters or greater.
```

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/methodtablebuilder.cpp:4181]: Missed that.

File: src/coreclr/utilcode/stgpooli.cpp [hunk 2]
```diff
@@ -212,7 +212,7 @@ HRESULT CPackedLen::SafeGetData(    // S_OK, or error
 
     BYTE const *pbData = reinterpret_cast<BYTE const*>(*ppData);
 
-    if (pbData + *pcbData < pbData)
+    if (*pcbData > reinterpret_cast<SIZE_T>(pDataSourceEnd) - reinterpret_cast<SIZE_T>(pDataSource))
```
Comment (davidwrighton) [inline src/coreclr/utilcode/stgpooli.cpp:215]: with this change, I think you can remove the followon check for data buffer overflow (and move that comment to this if statement.

File: src/coreclr/vm/customattribute.cpp [hunk 3]
```diff
@@ -1118,7 +1118,7 @@ BOOL COMCustomAttribute::CopyArrayVAL(BASEARRAYREF pArray, int nElements, BYTE *
     int sizeData;   // = size * 2; with integer overflow check
     if (!ClrSafeInt<int>::multiply(nElements, sizeof(T), sizeData))
         return FALSE;
-    if (*pBlob + sizeData < *pBlob)     // integer overflow check
+    if (sizeData > endBlob - *pBlob)     // integer overflow check
```
Comment (davidwrighton) [inline src/coreclr/vm/customattribute.cpp:1121]: I think this removes the need for the range check in the following line.

File: src/coreclr/utilcode/stgpooli.cpp (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/utilcode/stgpooli.cpp:215]: I don't think so, although initially I did do that. This only checks the range of the index, not how far it indexes into the data buffer. The check below ensures that starting from `pbData`, this index doesn't exceed the overall block.

File: src/coreclr/vm/customattribute.cpp (hunk 3 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/customattribute.cpp:1121]: This is the same logic as the previous comment. Checking the number itself versus checking the actual indexed memory.

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/customattribute.cpp:1121]: Oops. You're right here.

**Comment by AaronRobinsonMSFT** [comment]
@davidwrighton or @GrabYourPitchforks The CI is green. Any other concerns here?

File: src/coreclr/md/runtime/strongnameinternal.cpp [hunk 4]
```diff
@@ -290,7 +290,7 @@ HRESULT StrongNameTokenFromPublicKey(BYTE    *pbPublicKeyBlob,        // [in] pu
     // Check that the blob type is PUBLICKEYBLOB.
     pPublicKey = (PublicKeyBlob*) pbPublicKeyBlob;
 
-    if (pPublicKey->PublicKey + GET_UNALIGNED_VAL32(&pPublicKey->cbPublicKey) < pPublicKey->PublicKey) {
+    if (pPublicKey->cbPublicKey > cbPublicKeyBlob) {
```
Comment (davidwrighton) [inline src/coreclr/md/runtime/strongnameinternal.cpp:293]: This really should be using GET_UNALIGNED_VAL32.

Comment (AaronRobinsonMSFT) [inline src/coreclr/md/runtime/strongnameinternal.cpp:293]:
```suggestion
    if (GET_UNALIGNED_VAL32(&pPublicKey->cbPublicKey) > cbPublicKeyBlob) {
```

**Review by davidwrighton — APPROVED** [review]

**Pull request merged**